simulator/src/main/java/com/hazelcast/simulator/worker/TimeStepModel.java

```diff
@@ -0,0 +1,370 @@
+            }
+        }
+    }
+
+    private Map<Method, Double> loadProbabilities() {
+        Map<Method, Double> probabilities = new HashMap<Method, Double>();
+
+        double totalProbability = 0;
+        Method defaultMethod = null;
+        for (Method method : timeStepMethods) {
+            double probability = loadProbability(method);
+
+            if (probability == -1) {
+                if (defaultMethod != null) {
+                    throw new IllegalTestException("TimeStep method '" + method + "' can't have -1 probability. "
+                            + "method '" + defaultMethod + "' already has -1 probability and only one such method is allowed");
+                }
+                defaultMethod = method;
+            } else if (probability > 1) {
+                throw new IllegalTestException("TimeStep method '" + method + "' "
+                        + "can't have a probability larger than 1, found " + probability);
+            } else if (probability < 0) {
+                throw new IllegalTestException("TimeStep method '" + method + "' "
+                        + "can't have a probability smaller than 0, found " + probability);
+            } else {
+                totalProbability += probability;
+
+                if (totalProbability > 1) {
+                    throw new IllegalTestException("TimeStep method '" + method + "' with probability "
+                            + probability + " exceeds the total probability of 1");
+                }
+                probabilities.put(method, probability);
+            }
+        }
+
+        if (defaultMethod != null) {
+            double probability = 1 - totalProbability;
+            probabilities.put(defaultMethod, probability);
+        } else if (totalProbability < 1) {
+            throw new IllegalTestException("The total probability of timeStep methods in test " + testClass.getName()
+                    + " is smaller than 1. Found " + totalProbability);
+        }
+
+        return probabilities;
+    }
+
+    private double loadProbability(Method method) {
+        String propertyName = method.getName() + "Prob";
+        String configuredValue = propertyBinding.loadProperty(propertyName);
+        if (configuredValue == null) {
+            // nothing was specified. So lets use what is on the annotation
+            return method.getAnnotation(TimeStep.class).prob();
+        } else {
+            // the user has explicitly configured a probability
+            try {
+                return Double.parseDouble(configuredValue);
+            } catch (NumberFormatException e) {
+                throw new IllegalTestException(testClass.getName() + "." + propertyName
+                        + " value '" + configuredValue + "' is not a valid double value", e);
+            }
+        }
+    }
+
+    /**
+     * Returns null if there is one or no timestep methods.
```

Comment:
Should be `{@code null}`. Also `@return` should be completed or removed, right now this will be a warning (which sometimes cause trouble in the release, JavaDoc generation seems to be quite picky).